vllm/engine/arg_utils.py

```diff
@@ -498,27 +504,49 @@ def add_cli_args(parser: FlexibleArgumentParser) -> FlexibleArgumentParser:
                             '- 1K → 1024\n')
 
         # Guided decoding arguments
-        guided_decoding_kwargs = get_kwargs(DecodingConfig)
-        guided_decoding_group = parser.add_argument_group(
+        structured_output_kwargs = get_kwargs(StructuredOutputConfig)
+        structured_output_group = parser.add_argument_group(
             title="DecodingConfig",
-            description=DecodingConfig.__doc__,
+            description=StructuredOutputConfig.__doc__,
+        )
+        structured_output_group.add_argument(
```

Comment:
question for @hmellor:

For CLI do we want to add a `deprecated=True` to add_argument instead? right now I will just include notice in docstring

Reply:
Oh that's neat, yeah I'd prefer adding `deprecated=True` to the argument rather than changing the docstring

Reply:
I will open a separate PR for this then